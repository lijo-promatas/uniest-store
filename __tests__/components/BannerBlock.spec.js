import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import BannerBlock from '../../src/components/BannerBlock';

it('renders <BannerBlock /> component correctly', () => {
  const tree = renderer.create(<BannerBlock
    name="Banner block"
    items={[
      {
        url: 'test',
        main_pair: {
          icon: {
            http_image_path: '',
          },
        }
      }
    ]}
    onPress={() => {}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
