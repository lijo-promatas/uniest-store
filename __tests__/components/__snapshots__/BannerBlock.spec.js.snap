// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <BannerBlock /> component correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Banner block
  </Text>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      autoplay={false}
      autoplayDirection={true}
      autoplayTimeout={2.5}
      bounces={false}
      contentContainerStyle={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      contentOffset={
        Object {
          "x": 0,
        }
      }
      disableNextButton={false}
      height={200}
      horizontal={true}
      index={0}
      loadMinimal={false}
      loadMinimalSize={1}
      loop={true}
      onIndexChanged={[Function]}
      onMomentumScrollEnd={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollsToTop={false}
      showsButtons={false}
      showsHorizontalScrollIndicator={false}
      showsPagination={true}
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "height": 200,
                "width": 750,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "",
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
